src/js/common/schemaform/reducers/index.js

```diff
@@ -21,12 +21,19 @@ function recalculateSchemaAndData(initialState) {
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = _.mapValues('data', state);
```

Comment:
Just to be certain, I wanted to check the performance of this, so I wrote a little jsPerf using sample data from the form (captured from the redux store with `schema` and `uiSchema` still intact for added realism). This is what I found (note that I wasn't trying to compare the two, so I didn't optimize the native code at all):
![image](https://cloud.githubusercontent.com/assets/12970166/23975967/9fd89d6a-09a1-11e7-9cb9-401b0d85e3c2.png)
So while I definitely don't want to keep adding to the reducer code that runs every single keystroke (ack!), I think this addition isn't too bad.